ProcessLib/SmallDeformationWithLIE/SmallDeformationProcess.cpp

```diff
@@ -57,25 +57,57 @@ SmallDeformationProcess<DisplacementDim>::SmallDeformationProcess(
 {
     getFractureMatrixDataInMesh(mesh,
                                 _vec_matrix_elements,
+                                _vec_fracture_mat_IDs,
                                 _vec_fracture_elements,
                                 _vec_fracture_matrix_elements,
                                 _vec_fracture_nodes);
 
-    // set fracture property assuming a fracture forms a straight line
-    setFractureProperty(DisplacementDim,
-                        *_vec_fracture_elements[0],
-                        *_process_data._fracture_property.get());
+    if (_vec_fracture_mat_IDs.size() != _process_data._vec_fracture_property.size())
+        OGS_FATAL("The number of the given fracture properties (%d) are not consistent"
+                  " with the number of fracture groups in a mesh (%d).",
+                  _process_data._vec_fracture_property.size(),
+                  _vec_fracture_mat_IDs.size());
+
+    // create a map from a material ID to a fracture ID
+    auto max_frac_mat_id = std::max_element(_vec_fracture_mat_IDs.begin(),
+                                            _vec_fracture_mat_IDs.end());
```

Comment:
yes, it works, because the code accesses `_process_data._map_materialID_to_fractureID` only with material IDs of fractures.